src/hooks/helper/wallet.ts

```diff
@@ -265,10 +272,123 @@ export const checkIsNativeWallet = (selectedWallet: SupportWallet): boolean => {
   return hasProperty(supportWalletObj, selectedWallet);
 };
 
+const deleteWalletConnectDb = async (): Promise<void> => {
+  indexedDB.deleteDatabase('WALLET_CONNECT_V2_INDEXED_DB');
+  // Memo: wait for the DB to be deleted (above method doesn't return a promise)
+  await wait(2000);
+};
+
+interface IWcEthereumProvider {
+  provider: typeof WcEthereumProvider | undefined;
+  chainId: number;
+}
+
+const initWcProvider = async (): Promise<typeof WcEthereumProvider> => {
+  const astar = providerEndpoints[endpointKey.ASTAR];
+  const shiden = providerEndpoints[endpointKey.SHIDEN];
+  const shibuya = providerEndpoints[endpointKey.SHIBUYA];
+  const zKatana = providerEndpoints[endpointKey.ZKATANA];
+  const astarZkEvm = providerEndpoints[endpointKey.ASTAR_ZKEVM];
+
+  // Memo: this can be committed as it can be exposed on the browser anyway
+  const projectId = 'c236cca5c68248680dd7d0bf30fefbb5';
+
+  // Ref: https://docs.walletconnect.com/advanced/walletconnectmodal/options#explorerrecommendedwalletids-optional
+  // Memo: disabled 'desktop wallet' section as it doesn't work for our zkEVM wallet. We might want to filter the wallet by ids in the future
+  new WalletConnectModal({
+    projectId,
+    explorerRecommendedWalletIds: 'NONE',
+    // explorerExcludedWalletIds: 'ALL',
+  });
+
+  // Ref: https://docs.walletconnect.com/advanced/providers/ethereum
+  const provider = (await WcEthereumProvider.init({
+    projectId,
+    showQrModal: true,
+    optionalChains: [
+      Number(astar.evmChainId),
+      Number(shiden.evmChainId),
+      Number(shibuya.evmChainId),
+      // Number(astarZkEvm.evmChainId),
+      Number(zKatana.evmChainId),
+      EVM.SEPOLIA_TESTNET,
+    ],
+    chains: [EVM.ETHEREUM_MAINNET],
+    rpcMap: {
+      [astar.evmChainId]: astar.evmEndpoints[0],
+      [shiden.evmChainId]: shiden.evmEndpoints[0],
+      [shibuya.evmChainId]: shibuya.evmEndpoints[0],
+      // [astarZkEvm.evmChainId]: astarZkEvm.evmEndpoints[0],
+      [zKatana.evmChainId]: zKatana.evmEndpoints[0],
+      [String(EVM.SEPOLIA_TESTNET)]: String(rpcUrls[EVM.SEPOLIA_TESTNET][0]),
+    },
+  })) as any;
+
+  return provider;
+};
+
+export const initWalletConnectProvider = async (): Promise<IWcEthereumProvider> => {
+  const getProvider = async (): Promise<IWcEthereumProvider> => {
+    const provider = (await initWcProvider()) as any;
+
+    // Memo: remove the previous session in the wallet
+    // Ref: https://github.com/MyEtherWallet/MyEtherWallet/blob/52ea653a20f37becec7aa01f2564370f4366c8b1/src/modules/access-wallet/hybrid/handlers/WalletConnect/index.js#L135
+    const isConnected = await provider.connected;
+    if (isConnected) {
+      try {
+        await provider.disconnect();
+      } catch (error) {
+        console.error(error);
+      }
+    }
+
+    await provider.connect();
+    // Memo: to wait for syncing the correct chainId in provider
+    await wait(2000);
```

Comment:
Is there a way to achieve the same functionality without `wait` function. How you can be sure that in 2 seconds everything will be ready to continue code execution

Reply:
Sadly, I couldn't find another way. Without having this line, it returns `chainId: 1` even the wallet has connect to other chain.